Podverse/Playlist.swift

```diff
@@ -9,53 +9,145 @@
 import Foundation
 import CoreData
 
-class Playlist:NSManagedObject {
+class Playlist {
     
-    @NSManaged var id:String?
-    @NSManaged var podverseURL: String?
+    var id: String?
+    var title: String?
+    var slug: String?
+    var ownerId: String?
+    var ownerName: String?
+    var dateCreated: Date?
+    var lastUpdated: Date?
+    var isMyClips: Bool = false
+    var mediaRefs = [MediaRef]()
     
-    @NSManaged var ownerId:String
-    @NSManaged var ownerName:String?
+    static func jsonToPlaylist(item: [String:Any]) -> Playlist {
     
-    @NSManaged var title: String?
-    
-    @NSManaged var dateCreated: NSDate?
-    @NSManaged var lastUpdated: NSDate?
-    
-    @NSManaged var sharePermission: String?
+        let playlist = Playlist()
+        
+        playlist.id = item["id"] as? String
+        playlist.isMyClips = item["isMyClips"] as? Bool ?? false
+        playlist.slug = item["slug"] as? String
+        playlist.title = item["title"] as? String
+        playlist.ownerId = item["ownerId"] as? String
+        playlist.ownerName = item["ownerName"] as? String
+        
+        if let lastUpdated = item["lastUpdated"] as? String {
+            playlist.lastUpdated = lastUpdated.toServerDate()
+        }
+        
+        if let dateCreated = item["dateCreated"] as? String {
+            playlist.dateCreated = dateCreated.toServerDate()
+        }
+        
+        if let mediaRefsJSON = item["mediaRefs"] as? [[String:Any]] {
+            for item in mediaRefsJSON {
+                let mediaRef = MediaRef.jsonToMediaRef(item: item)
+                playlist.mediaRefs.append(mediaRef)
+            }
+        }
     
-    @NSManaged var isMyEpisodes: Bool
-    @NSManaged var isMyClips: Bool
+        return playlist
     
-    @NSManaged var episodes: Set<Episode>?
+    }
     
-    var allItems: [Any] {
-        get {
-            var allItemsArray: [Any] = []
-//            if let episodes = episodes {
-//                for episode in episodes {
-//                    allItemsArray.append(episode)
-//                }
-//            }
-//            if let clips = clips {
-//                for clip in clips {
-//                    allItemsArray.append(clip)
-//                }
-//            }
+    static func retrievePlaylistFromServer(id: String, completion: @escaping (_ playlist: Playlist?) -> Void) {
+        
+        if let url = URL(string: "http://localhost:8080/api/playlist") {
```

Comment:
Suggestion for this URL (and possibly in other locations as well):
Have 3 constants in the Constants file:
```
let DEV_URL = "http://localhost:8080/api/"
let PROD_URL = "whatever the non localhost url will be"
let BASE_URL = DEV_URL
```
Then replace all places in the app with a url with the BASE_URL. This way, when we need to switch urls, we only need to switch it out it in the constants file and set the BASE_URL to whatever we want. And use it as such: `URL(string: BASE_URL + "playlist")`